src/Lykke.Service.PaySettlement.Services/TransferToMarketService.cs

```diff
@@ -0,0 +1,176 @@
+using Lykke.Service.PaySettlement.Core.Domain;
+using Lykke.Service.PaySettlement.Core.Services;
+using Lykke.Service.PaySettlement.Core.Settings;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Timers;
+using PaymentRequestStatus = Lykke.Service.PayInternal.Contract.PaymentRequest.PaymentRequestStatus;
+
+namespace Lykke.Service.PaySettlement.Services
+{
+    public class TransferToMarketService : IStartable, IStopable, ITransferToMarketService
+    {
+        private readonly ITransferToMarketQueue _transferToMarketQueue;        
+        private readonly IPayInternalClient _payInternalClient;
+        private readonly IPayMerchantClient _payMerchantClient;
+        private readonly IPaymentRequestsRepository _paymentRequestsRepository;
+        private readonly IStatusService _statusService;
+        private readonly TransferToMarketServiceSettings _settings;
+        private readonly ILog _log;
+        private readonly Timer _timer;
+
+        public TransferToMarketService(ITransferToMarketQueue transferToMarketQueue,
+            IPayInternalClient payInternalClient, IPayMerchantClient payMerchantClient, 
+            IPaymentRequestsRepository paymentRequestsRepository, IStatusService statusService,
+            TransferToMarketServiceSettings settings, ILogFactory logFactory)
+        {
+            _transferToMarketQueue = transferToMarketQueue;
+            _payInternalClient = payInternalClient;
+            _payMerchantClient = payMerchantClient;
+            _paymentRequestsRepository = paymentRequestsRepository;
+            _statusService = statusService;
+            _settings = settings;
+            _log = logFactory.CreateLog(this);
+            _timer = new Timer(settings.Interval.TotalMilliseconds);            
+        }
+
+        public async Task AddToQueueIfSettlement(IPaymentRequest paymentRequest)
+        {
+            if (paymentRequest.Status != PaymentRequestStatus.Confirmed)
+            {
+                return;
+            }
+
+            if (!string.Equals(paymentRequest.PaymentAssetId, _settings.PaymentAssetId,
+                StringComparison.OrdinalIgnoreCase))
+            {
+                _log.Info($"Skip payment request because payment assetId is {paymentRequest.PaymentAssetId}.",
+                    new {PaymentRequestId = paymentRequest.Id});
+                return;
+            }
+
+            if (!_settings.SettlementAssetIds.Contains(paymentRequest.SettlementAssetId,
+                StringComparer.OrdinalIgnoreCase))
+            {
+                _log.Info($"Skip payment request because settlement assetId is {paymentRequest.PaymentAssetId}.",
+                    new {PaymentRequestId = paymentRequest.Id});
+                return;
+            }
+
+            MerchantModel merchant = await _payMerchantClient.Api.GetByIdAsync(paymentRequest.MerchantId);
+            if (merchant == null)
```

Comment:
This condition will never be true, the client will raise an exception with HttpStatusCode = NotFound.